packages/dds/tree/src/feature-libraries/sequence-field/markQueue.ts

```diff
@@ -85,7 +77,14 @@ export class MarkQueue<T> {
     public dequeueInput(length: number): InputSpanningMark<T> {
         const mark = this.dequeue();
         assert(isInputSpanningMark(mark), 0x4e3 /* Can only split sized marks on input */);
-        const [mark1, mark2] = splitMarkOnInput(mark, length, this.genId, this.moveEffects);
+        const [mark1, mark2] = splitMarkOnInput(
+            mark,
+            this.revision,
+            length,
+            this.genId,
+            this.moveEffects,
+            !this.consumeEffects,
```

Comment:
Shouldn't this be `false` instead of `!this.consumeEffects`?
IOW, we don't want to consume effects here even if `this.dequeue()` didn't consume them, right?

Reply:
Same for `dequeueOutput`.

Reply:
Maybe I should start by asking: why would we ever want to pass in `true` here?
If `this.consumeEffects` is true then the `tryDequeue` will already have consumed the effects.
If `this.consumeEffects` is true then why are we consuming the effects here? Doesn't that contradict the wishes of the caller of the constructor?

Reply:
I was reading the parameter wrong on splitMarkOnInput. Resolving this thread.